tree/tree.go

```diff
@@ -0,0 +1,160 @@
+package tree
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/vocdoni/arbo"
+	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/db/badgerdb"
+)
+
+// Tree defines the struct that implements the MerkleTree functionalities
+type Tree struct {
+	t  *arbo.Tree
+	db db.Database
+}
+
+// Options is used to pass the parameters to load a new Tree
+type Options struct {
+	// Path of the db.Database to be used
+	Path string
+	// MaxLevels that will have the Tree
+	MaxLevels int
+	// HashFunc defines the hash function that the tree will use
+	HashFunc arbo.HashFunction
+}
+
+// New returns a new Tree, if there already is a Tree in the database, it will
+// load it
+func New(opts Options) (*Tree, error) {
+	database, err := badgerdb.New(badgerdb.Options{Path: opts.Path})
+	if err != nil {
+		return nil, err
+	}
+
+	tree, err := arbo.NewTree(database, opts.MaxLevels, opts.HashFunc)
```

Comment:
I'm thinking that maybe we will need the `arbo.NewTree` to optionally accept a writeTx.  Like `arbo.NewTreeWithTx`.  The use case will be the StateDB.  Imagine in two different transactions in the same block you add two leafs in a new tree.  If you open the tree from the DB, the root will not exists, but in the StateDB tx after the first transaction the root already exists.